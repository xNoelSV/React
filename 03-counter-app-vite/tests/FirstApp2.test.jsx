/**
* @jest-environment jsdom
*/

const { render, screen } = require("@testing-library/react")
const { FirstApp } = require("../src/FirstApp")

describe('Pruebas en <FirstApp /> 2', () => {

  const title = 'Hola, soy Noel';
  const subTitle = 'Soy un subtitulo';

  test('Debe de hacer match con el snapshot', () => {
    
    const { container } = render( <FirstApp title={title} /> )
    expect( container ).toMatchSnapshot();

  })

  test('Debe de mostrar el mensale "Hola, soy Noel"', () => {

    render( <FirstApp title={ title }/> );
    screen.debug();
    expect( screen.getByText(title) ).toBeTruthy();

  }) 
  
  test('Debe de mostrar el tÃ­tulo en un h1', () => {
    
    render( <FirstApp title={ title }/> );
    expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );

  })

  test('Debe de mostrar el subtitulo enviado por props', () => {
    
    render( 
      <FirstApp
        title={ title }
        subTitle={ subTitle }
      />
    )

    expect( screen.getAllByText(subTitle).length ).toBe(3);

  })

})
