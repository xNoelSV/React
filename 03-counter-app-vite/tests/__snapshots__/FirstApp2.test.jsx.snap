// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> 2 Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Noel
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Noel Sari√±ena
  </p>
</div>
`;
